conda/trust/signature_verification.py

```diff
@@ -193,23 +193,27 @@ def _fetch_channel_signing_data(
             headers["If-Modified-Since"] = mod_stamp
 
         try:
-            # The `auth` argument below looks a bit weird, but passing `None` seems
-            # insufficient for suppressing modifying the URL to add an Anaconda
-            # server token; for whatever reason, we must pass an actual callable in
-            # order to suppress the HTTP auth behavior configured in the session.
+            # Assume trust metadata is intended to be "generally available",
+            # and specifically, _not_ protected by a conda/binstar token.
+            # Seems reasonable, since we (probably) don't want the headaches of
+            # dealing with protected, per-channel trust metadata.
             #
-            # TODO: Figure how to handle authn for obtaining trust metadata,
-            # independently of the authn used to access package repositories.
+            # Note: Setting `auth=None` here does allow trust metadata to be
+            # protected using standard HTTP basic auth mechanisms, with the
+            # login information being provided in the user's netrc file.
+            saved_token_setting = context.add_anaconda_token
+            context.add_anaconda_token = False
```

Comment:
I worry a little about thread-safety here, since context is just a module attribute and behind `self.session` is the LRU cached instance of the `CondaSession` class, which behind the scenes runs the requests/urllib3 machinery, but is using a thread local to store session instances per thread.

E.g. in case an exception occurs during `self.session.get`, it won't reset `context.add_anaconda_token` again, which may or may not have issues downline when it arrives at the exception handler. Would a 

```python
saved_token_setting = context.add_anaconda_token
context.add_anaconda_token = False
try:
    ...
finally:
    context.add_anaconda_token = saved_token_setting
```

Reply:
Ooohhh...good catch. Modified the PR to add the suggested `finally` block (and to remove the basically pointless `except`).